src/main/java/de/cyklon/reflection/entities/impl/ClassFileImpl.java

```diff
@@ -0,0 +1,61 @@
+package de.cyklon.reflection.entities.impl;
+
+import de.cyklon.reflection.entities.ClassFile;
+import de.cyklon.reflection.entities.OfflinePackage;
+import de.cyklon.reflection.entities.ReflectClass;
+import de.cyklon.reflection.exception.NotFoundException;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class ClassFileImpl implements ClassFile {
+
+	private final String className;
+
+	ClassFileImpl(String className) {
+		this.className = className;
+		if (checkClass(className)) throw new NotFoundException(className, "class", "");
+	}
+
+	private static boolean checkClass(@NotNull String className) {
```

Comment:
The return value of this method is confusing, it should return true if it is successful and false if it fails